cloud/scope/machine.go

```diff
@@ -744,7 +744,7 @@ func (m *MachineScope) getCompartmentId() string {
 
 func (m *MachineScope) getGetControlPlaneMachineSubnet() *string {
 	for _, subnet := range m.OCIClusterAccessor.GetNetworkSpec().Vcn.Subnets {
-		if subnet.Role == infrastructurev1beta2.ControlPlaneRole {
+		if subnet != nil && subnet.ID != nil && subnet.Role == infrastructurev1beta2.ControlPlaneRole {
```

Comment:
Let me ask this. If ID is `nil` that doesn't matter here right? This function returns `nil` anyway. To keep the code simpler what if we removed `&& subnet.ID != nil`

```suggestion
		if subnet != nil && subnet.Role == infrastructurev1beta2.ControlPlaneRole {
```

Reply:
I won't make this on ever other line like this, but If you agree with this idea lets do that other places as well.